ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/conditions/PostConditions.java

```diff
@@ -75,29 +77,46 @@ public void verify(
   }
 
   public Optional<PostConditionsNode> asNode(
-      final List<PostTopicNode> additional
+      final List<PostTopicNode> additionalTopics,
+      final List<SourceNode> additionalSources
   ) {
-    if (this == NONE && additional.isEmpty()) {
+    if (this == NONE && additionalTopics.isEmpty() && additionalSources.isEmpty()) {
       return Optional.empty();
     }
 
+    return Optional.of(new PostConditionsNode(
+        withAdditionalSources(additionalSources),
+        Optional.of(withAdditionalTopics(additionalTopics))
+    ));
+  }
+
+  private List<SourceNode> withAdditionalSources(final List<SourceNode> additionalSources) {
```

Comment:
It's a bit odd to combine the additional sources (list of all sources from metastore) with the test-provided post condition sources, since the former should be a superset of the latter or else the test will fail. This is the existing pattern for collecting topics into the post conditions, though, so I've maintained it here. cc @big-andy-coates for context on why topics are aggregated like this.

Reply:
What you say sounds sensible / right.  If the we're capturing all sources then we don't need to maintain any that have been added in the test.

Same sounds true to topics.

Reply:
> Maybe just debug / hack some code in to confirm its always a super set.

I'm not convinced this is worth it. Suppose someone writes a bad test with a topic or source in the post conditions that exists but does not validate properly (e.g., expecting a source with a particular name to be a table when it should be stream). Unless the superset check is nested (i.e., checks that fields are equal up to optionality), the code for generating historic plans will overwrite the bad test-provided post condition and the fact that the test fails on master will still be hidden in the historic plan. In other words, adding a partial superset check still leaves gaps, so I think it's better to blindly overwrite the topics and sources when generating historic plans and leave catching bad test files to QTT instead.

I'll update the code here to overwrite topics and sources rather than trying to merge them with test-provided post conditions.

Reply:
Ah, I think I misunderstood your suggestion earlier. I think 

> Maybe just debug / hack some code in to confirm its always a super set.

means to sanity check offline that for existing tests, the set of all topics and sources is always a superset of what's specified in the post conditions, and then simplify the code to just replace the topics and sources with the lists of all topics and all sources. If so, this aligns with what I wrote above as well. I'll do this.

Reply:
Confirmed that the sets of all topics and sources are always supersets of the test-provided topics and sources from the post conditions, and removed the dead code. Had to add another commit for regenerating specs since the order of the topics/sources in the specs has changed in many cases, though there are no substantive changes.